src/devices/bmx055.c

```diff
@@ -0,0 +1,74 @@
+#include "devices/bmx055.h"
+#include "stm32f3hal/gpio.h"
+#include "stm32f3hal/i2c.h"
+
+static GPIO_PIN_Config bmx055_pow_config = {
```

Comment:
It may be worth moving these configs to their own file, especially the GPIO ones. Reasoning: If you need to make changes to the board later, or want to port this code to another board, the diff will make more sense to have deltas in something like `PLATFORM_NAME_ioconfig.c` instead of in a bunch of random headers. Not necessarily worthwhile unless there will be other IOs configured on your board.